pkg/sdn/plugin/networkpolicy.go

```diff
@@ -0,0 +1,362 @@
+}
+
+func (np *networkPolicyPlugin) DeleteNetNamespace(netns *osapi.NetNamespace) {
+	np.lock.Lock()
+	defer np.lock.Unlock()
+
+	delete(np.namespaces, netns.NetID)
+}
+
+func (np *networkPolicyPlugin) GetVNID(namespace string) (uint32, error) {
+	return np.vnids.WaitAndGetVNID(namespace)
+}
+
+func (np *networkPolicyPlugin) GetNamespaces(vnid uint32) []string {
+	return np.vnids.GetNamespaces(vnid)
+}
+
+func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
+	inUse := npns.refs > 0
+	if !inUse && !npns.inUse {
+		return
+	}
+
+	glog.V(5).Infof("syncNamespace %d", npns.vnid)
+	otx := np.node.ovs.NewTransaction()
+	otx.DeleteFlows("table=80, reg1=%d", npns.vnid)
+	if inUse {
+		if npns.isolated {
+			for _, npp := range npns.policies {
+				for _, flow := range npp.flows {
+					otx.AddFlow("table=80, priority=100, reg1=%d, %s actions=output:NXM_NX_REG2[]", npns.vnid, flow)
+				}
+			}
+		} else {
+			otx.AddFlow("table=80, priority=100, reg1=%d, actions=output:NXM_NX_REG2[]", npns.vnid)
+		}
+	}
+	if err := otx.EndTransaction(); err != nil {
+		glog.Errorf("Error syncing OVS flows for VNID: %v", err)
+	}
+	npns.inUse = inUse
+}
+
+func (np *networkPolicyPlugin) RefVNID(vnid uint32) {
+	np.lock.Lock()
+	defer np.lock.Unlock()
+
+	npns, exists := np.namespaces[vnid]
+	if !exists {
+		return
+	}
+
+	npns.refs += 1
+	np.syncNamespace(npns)
+}
+
+func (np *networkPolicyPlugin) UnrefVNID(vnid uint32) {
+	np.lock.Lock()
+	defer np.lock.Unlock()
+
+	npns, exists := np.namespaces[vnid]
+	if !exists {
+		return
+	}
+	if npns.refs == 0 {
+		glog.Warningf("refcounting error on vnid %d", vnid)
+		return
+	}
+
+	npns.refs -= 1
+	np.syncNamespace(npns)
+}
+
+func (np *networkPolicyPlugin) parseNetworkPolicy(npns *npNamespace, policy *extensions.NetworkPolicy) (*npPolicy, error) {
+	npp := &npPolicy{policy: *policy}
+	allowAll := false
+
+	for _, rule := range policy.Spec.Ingress {
+		if len(rule.Ports) == 0 && len(rule.From) == 0 {
+			allowAll = true
+			break
+		}
+	}
+
+	if allowAll {
+		npp.flows = []string{""}
+	} else {
+		sort.Strings(npp.flows)
+	}
+	glog.V(5).Infof("Parsed NetworkPolicy: %#v", npp)
+	return npp, nil
+}
+
+func (np *networkPolicyPlugin) updateNetworkPolicy(npns *npNamespace, policy *extensions.NetworkPolicy) bool {
+	npp, err := np.parseNetworkPolicy(npns, policy)
+	if err != nil {
+		glog.Infof("Unsupported NetworkPolicy %s/%s (%v); treating as deny-all", policy.Namespace, policy.Name, err)
+		npp = &npPolicy{policy: *policy}
+	}
+
+	oldNPP, existed := npns.policies[policy.UID]
+	npns.policies[policy.UID] = npp
+
+	changed := !existed || !reflect.DeepEqual(oldNPP.flows, npp.flows)
+	if !changed {
+		glog.V(5).Infof("NetworkPolicy %s/%s is unchanged", policy.Namespace, policy.Name)
+	}
+	return changed
+}
+
+func (np *networkPolicyPlugin) watchNetworkPolicies() {
+	RunEventQueue(np.node.kClient.ExtensionsClient.RESTClient(), NetworkPolicies, func(delta cache.Delta) error {
```

Comment:
The "EventQueue" used in pkg/sdn/plugin is a wrapper around DeltaFIFO (#10070) not the old pkg/client/cache/eventqueue.go.